src/net_processing.cpp

```diff
@@ -2630,6 +2760,46 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         return;
     }
 
+    // Received from a peer demonstrating readiness to announce transactions via reconciliations.
+    // This feature negotiation should happen:
+    // - between VERSION and VERACK to avoid relay problems from switching annoyncement protocols
+    //   after the connection is up
+    // - after WTXID because we reconcile only with peers supporting wtxid-relay
+    if (msg_type == NetMsgType::SENDRECON) {
+        if (pfrom.fSuccessfullyConnected) {
+            // Disconnect peers that send a SENDRECON message after VERACK.
+            LogPrint(BCLog::NET, "sendrecon received after verack from peer=%d; disconnecting\n", pfrom.GetId());
+            pfrom.fDisconnect = true;
+            return;
+        }
+
+        if (!pfrom.m_tx_relay) {
+            // If we chose a peer to not send us transactions, disconnect if they want to reconcile.
+            LogPrint(BCLog::NET, "sendrecon received from non-tx-relay peer=%d; disconnecting\n", pfrom.GetId());
+            pfrom.fDisconnect = true;
+            return;
+        }
+
+        LOCK(cs_main);
+        if (!State(pfrom.GetId())->m_wtxid_relay) {
+            // Disconnect peers that send a SENDRECON message before/without WTXIDRELAY.
```

Comment:
>I think we should update BIP330 to mark the dependency on BIP339.

Agree.

>Also, should we bump the protocol version ? Otherwise a bip330 peer will send SENDRECON for nothing to bip339-only peer (core version 20+).

I don't understand what you mean here.

Reply:
A bip330 peer sends a SENDRECON message to a wtxid-relay peer but not supporting bip330. 
The wtxid-relay peer doesn't understand SENDRECON and will ignore the message.
The bip330 wasted SENDRECON message bandwidth.

If the bip330 peer would have been aware that the wtxid-relay peer didn't support bip330 it could have avoid sending SENDRECON. Right now, I think your check implies that wtxid-relay peers also support bip330, I believe a protocol version bump would avoid this minor issue.